core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -402,7 +402,12 @@ private[spark] class Executor(
           // Don't forcibly exit unless the exception was inherently fatal, to avoid
           // stopping other tasks unnecessarily.
           if (Utils.isFatalError(t)) {
-            SparkUncaughtExceptionHandler.uncaughtException(t)
+            if (!isLocal) {
+              Thread.getDefaultUncaughtExceptionHandler.
```

Comment:
So even for fatal error like OOME or other fatal ones the executor will try to remain up in this case. Should we have some more cases for fatal exceptions here?